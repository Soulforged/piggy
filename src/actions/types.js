//@flow

import type { Region } from 'src/Types';

export type ChangePosition = {
  position: Region
};

export type SetHasError = {
  hasError: boolean
};

//FIXME use Symbols maybe
export const LOGIN = 'LOGIN';
export const CHANGE_POSITION = 'CHANGE_POSITION';
export const SET_HAS_ERROR = 'SET_HAS_ERROR';
