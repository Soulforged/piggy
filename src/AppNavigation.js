//@flow
import { StackNavigator } from 'react-navigation';
import LoginNavigation from './LoginNavigation';
import MainNavigation from './MainNavigation';

const routes = {
    Login: { screen: LoginNavigation },
    Main: { screen: MainNavigation }
  },
  config = {
    headerMode: 'none',
    initialRouteName: 'Login'
  };

const AppNavigation = StackNavigator(routes, config);
export default AppNavigation;
