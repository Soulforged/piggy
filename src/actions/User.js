//@flow

import type { Region } from 'src/Types';
import { LOGIN, CHANGE_POSITION, SET_HAS_ERROR } from './types';

export const login = () => ({ type: LOGIN });
export const changePosition = (position: Region) => ({ type: CHANGE_POSITION, position });
export const setHasError = (hasError: boolean) => ({ type: SET_HAS_ERROR, hasError });
