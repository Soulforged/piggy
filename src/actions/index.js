//@flow
import { createActions } from 'redux-actions';

export default createActions({
  LOGIN: () => ({}),
  SET_ERROR: (error: Error | false) => error,
});
