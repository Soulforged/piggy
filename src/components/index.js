//@flow

import ErrorBoundary from './containers/ErrorBoundaryContainer';
import Map from './containers/MapContainer';

export { ErrorBoundary, Map };
