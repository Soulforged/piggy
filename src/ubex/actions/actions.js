//@flow
import { createActions } from 'redux-actions';

import type { Region } from 'ubex/types';

const receivePredictions = (json: Object) => ({
  predictions: json.predictions.map( ({place_id, description}) => (
    { id: place_id, description, key: place_id }
  )),
  lastUpdated: Date.now()
});

export default createActions({
  CHANGE_POSITION: (position: Region) => ({ position }),
  REQUEST_PREDICTIONS: (description: string) => ({ description }),
  RECEIVE_PREDICTIONS_ERROR: (error: string) => ({ error }),
  RECEIVE_PREDICTIONS: receivePredictions,
  REQUEST_PLACE_DETAILS: (id: string) => ({ id }),
  RECEIVE_PLACE_DETAILS_ERROR: (error: string) => ({ error }),
  RECEIVE_PLACE_DETAILS: (json: Object) => json.result,
  SHOW_PREDICTIONS: () => ({}),
});
