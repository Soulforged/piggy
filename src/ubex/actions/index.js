//@flow
import { createActions } from 'redux-actions';

import type { Region } from 'ubex/Types';

export default createActions({
  CHANGE_POSITION: (position: Region) => ({ position })
});
