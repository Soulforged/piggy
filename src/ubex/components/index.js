//@flow

import Map from './containers/MapContainer';

export { Map };
