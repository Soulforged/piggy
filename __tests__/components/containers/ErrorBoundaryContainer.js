//@flow

import React from 'react';
import { View } from 'react-native';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import createStore from 'src/reduxconf';

import actions from 'src/actions';
import { ErrorBoundary } from 'src/components';

const { setHasError } = actions;
const store = createStore();

const component = () => (
  <Provider store={store} >
    <ErrorBoundary>
      <View />
    </ErrorBoundary>
  </Provider>
);

it('renders succesfully', () => {
  const rendered = renderer.create(component()).toJSON();
  expect(rendered).toMatchSnapshot();
});

it('calls setHasError action on catch', () => {
  const rendered = renderer.create(component()).toJSON();
  store.dispatch(setHasError(true));
  expect(rendered).toMatchSnapshot();
});
