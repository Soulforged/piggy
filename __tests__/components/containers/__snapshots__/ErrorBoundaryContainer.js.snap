// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`calls setError action on catch 1`] = `
<View>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
  >
    message
  </Text>
  <View
    accessibilityComponentType="button"
    accessibilityLabel={undefined}
    accessibilityTraits={
      Array [
        "button",
      ]
    }
    accessible={true}
    collapsable={undefined}
    hasTVPreferredFocus={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={undefined}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#007AFF",
              "fontSize": 18,
              "padding": 8,
              "textAlign": "center",
            },
          ]
        }
      >
        Close
      </Text>
    </View>
  </View>
</View>
`;

exports[`renders succesfully 1`] = `<View />`;
